### PR TITLE
Draft: MAS UI/Console Installer

### PR DESCRIPTION
**Pull request by b1zzu**
**Architecture**
https://lucid.app/lucidchart/46b90b22-9b5b-4940-a06e-580d3b589b38/edit?invitationId=inv_742bbb55-3ce9-405a-8afa-f9146ef2b666 

image

**Known Issue:**
Is impossible to create Kafka instances from the UI because the /kafkas_mgmt/v1/cloud_providers endpoint doesn't work

**Missing components:**
- kas-ui
- kafka-ui
- guides

### CODE REVIEW

**Comment by MikeEdgar** [comment]
@b1zzu do the changes to use QA SSO rather than a local MAS-SSO in any way help with this PR? I have an upcoming change to the admin API server to change the back-end to Quarkus and I'm interested in using the UI to catch regressions ahead of the stage environment.

**Comment by b1zzu** [comment]
Hi @MikeEdgar, there is just more work than it needs to go in this PR to make the UI usable, the issues I'm facing now are:
- The UI relies on the AMS API to check the terms and services which means that is not possible to create Kafka instances from the UI 
- The Kafka Admin API is insecure on HTTP, and this is also doesn't work out of the box with the UI therefore right now is not possible to edit/update/delete ACLS/topics/groups from the UI
- Service account creation as the list is not working too because is blocked by CORS policy

In conclusion, although the UI can be opened, and is possible to login, some of the most critical functionalities are not working.

**Comment by b1zzu** [comment]
@MikeEdgar I have also opened a PR to make the admin server URL configurable so that I can use HTTP:// in the kas-installer: https://github.com/redhat-developer/app-services-ui/pull/375

**Comment by rareddy** [comment]
@wtrocki fyi..

**Comment by wtrocki** [comment]
This is good investigation but there are couple concerns:

- How reliable we want to be with this? Can we accept some hacks on clients to make it work? 
- Can we run local proxy on every client to use localhost in the UI? That is what is typically done in such scenarios with redirect URI issues. Localhost redirect will also help us to setup proper local development setup.
- Runnning whole app (app-services-ui) does run not only kafka. There is shared logic for service registry etc. Should we run federated modules instead?
- Some issues we see might require us to wait - mas sso migration will help. We can also try to reduce reliance on mas-sso
- Big problem right now is lack of good local developement setup for our Uis. Users need vpn to run it. We should possibly fix that first as it would make kas-installer integration trivial

**Comment by wtrocki** [comment]
While I'm impressed by @b1zzu work on this I think we should discuss number of steps we need to do before integrating with kas-installer. This holefuly will make architecure simpler (with only component that we would need to run is UI proxy:

kas-installer start-ui

Anyone using minikube should be already familliar with this concept. It think that issues we seeing uncovering our strong reliance on product APIs in mixed open source scenario. Some work should happen to make our AMS and Keycloak integration pluggable if we looking to have end to end deployment model to any cluster

**Comment by MikeEdgar** [comment]
Replaced with functionality in #197
